### ISSUE TITLE
[question] Empty/Blank/Null Value for singleSelect

### ISSUE THREAD

**Issue by tnoetzel** [mui/mui-x#5996]
### Order ID 💳

49215

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

I'm trying to allow users to set a null or "empty" value for a column of type `singleSelect`.

If my `valueOptions` array has an item of `{ value: "", label: "None" }`, then users _can_ set the value back to empty, _but_ having this option causes a `Warning: Encountered two children with the same key` error when trying to filter on that column, because filters default to adding an empty option:

<img width="594" alt="two-empty-options" src="https://user-images.githubusercontent.com/1002744/187966451-2213e022-b712-4306-a5f3-71f55a045c41.png">

How can I allow users to select an empty option or clear the column via the select dropdown without causing this error on the filter?

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: macOS 12.3.1
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: Not Found
    npm: 8.9.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Not Found
    Firefox: 101.0.1
    Safari: 15.4
  npmPackages:
    @emotion/react: ^11.6.0 => 11.6.0 
    @emotion/styled: ^11.6.0 => 11.6.0 
    @mui/base:  5.0.0-alpha.92 
    @mui/icons-material: ^5.6.2 => 5.6.2 
    @mui/material: ^5.1.0 => 5.10.0 
    @mui/private-theming:  5.9.3 
    @mui/styled-engine:  5.10.0 
    @mui/system:  5.10.0 
    @mui/types:  7.1.5 
    @mui/utils:  5.9.3 
    @mui/x-data-grid:  5.15.2 
    @mui/x-data-grid-pro: ^5.15.2 => 5.15.2 
    @mui/x-license-pro:  5.15.0 
    @types/react: ^17.0.20 => 17.0.20 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2
```
</details>

**Comment by m4theushw** [comment]
Can you use `{ value: "None", label: "None" }` instead of `""` and treat `None` as empty every time it's used? An empty string is kind of a reserved value for the grid. Even if we allow to customize the label for the empty option it would still cause problems because we can't determine if a filter item has no value set or if it's value is `""`.

**Comment by tnoetzel** [comment]
Yeah, that's definitely _possible_, but it feels like a hack for a pretty common use case. IMO, there should be an option on `singleSelect` to include an empty value _or_ the filters should automatically ignore empty values and not include them.
